cmd/git-sync/main.go

```diff
@@ -688,6 +688,12 @@ func addWorktreeAndSwap(ctx context.Context, gitRoot, dest, branch, rev string,
 		return err
 	}
 
+	// Check for rev after fetch
+	_, err = localHashForRev(ctx, rev, gitRoot)
```

Comment:
The local ref hash won't be updated after fetch. I didn't see how this check will prevent a sync failure.

Besides checking the hash, we need to make sure the old worktree is deleted so that it won't block a new sync.

Reply:
> It would also be valuable to delete the worktree for a hash after an error. The fact we don't clean up the worktree if we can't git reset --hard means the next time it syncs (without a new change) it'll fail because it can't create the worktree with the same name

https://github.com/kubernetes/git-sync/pull/414

Reply:
At this point the worktree has not been created.

I did not understand the comment about local ref hash wont be updated after fetch. 
This function does a git rev-parse on the local git. If that rev doesnt exist (race condition?) we return gracefully and wait for the next sync.

Reply:
The function executes `git rev-parse HEAD` on the local repo, which returns the commit revision ID that HEAD is pointing to. The rev is stored locally, it should always exist.

Reply:
I do not understand this - it runs the re-parse but doesn't actually check the result?  What am I missing - this seems like a noop...

Reply:
@nan-yu : Thanks. I understand. Rechecking the code.

Reply:
I have updated code and added a test